framework/BaseYii.php

```diff
@@ -345,23 +345,23 @@ public static function createObject($type, array $params = [])
             return static::$container->get($type, $params);
         }
 
-        if (is_array($type) && isset($type['class'])) {
-            $class = $type['class'];
-            unset($type['class']);
-            return static::$container->get($class, $params, $type);
-        }
-
-        if (is_array($type) && isset($type['__class'])) {
-            $class = $type['__class'];
-            unset($type['__class']);
-            return static::$container->get($class, $params, $type);
-        }
-
         if (is_callable($type, true)) {
             return static::$container->invoke($type, $params);
         }
 
         if (is_array($type)) {
+            if (isset($type['__class'])) {
+                $class = $type['__class'];
+                unset($type['__class'], $type['class']);
+                return static::$container->get($class, $params, $type);
+            }
+
+            if (isset($type['class'])) {
+                $class = $type['class'];
+                unset($type['class']);
+                return static::$container->get($class, $params, $type);
+            }
+
```

Comment:
Please use `if (!is_array($type)) {` then throw an exception

Reply:
`$type` here exactly is an array

Reply:
I mean it needs to flip over the condition from `if (is_array($type)) {` to `if (NOT is_array($type)) {` then throw an exception, that `$type` is not an array and continue to check a properties

Reply:
We don't need to change the condition here because this exception is thrown only on these three conditions matching:

- the config **is** an array
- there is no `class` key in it
- there is no `__class` key in it